contracts/BC_fusion/StBNB.sol

```diff
@@ -0,0 +1,347 @@
+        return true;
+    }
+
+    /**
+     * @notice Atomically decreases the allowance granted to `_spender` by the caller by `_subtractedValue`.
+     *
+     * This is an alternative to `approve` that can be used as a mitigation for
+     * problems described in:
+     * https://github.com/OpenZeppelin/openzeppelin-contracts/blob/b709eae01d1da91902d06ace340df6b324e6f049/contracts/token/ERC20/IERC20.sol#L57
+     * Emits an `Approval` event indicating the updated allowance.
+     *
+     * Requirements:
+     *
+     * - `_spender` cannot be the zero address.
+     * - `_spender` must have allowance for the caller of at least `_subtractedValue`.
+     */
+    function decreaseAllowance(address _spender, uint256 _subtractedValue) external returns (bool) {
+        address owner = _msgSender();
+        uint256 currentAllowance = allowance(owner, _spender);
+        require(currentAllowance >= _subtractedValue, "ALLOWANCE_BELOW_ZERO");
+        unchecked {
+            _approve(owner, _spender, currentAllowance - _subtractedValue);
+        }
+        return true;
+    }
+
+    /**
+     * @return the total amount of shares in existence.
+     *
+     * @dev The sum of all accounts' shares can be an arbitrary number, therefore
+     * it is necessary to store it in order to calculate each account's relative share.
+     */
+    function getTotalShares() external view returns (uint256) {
+        return _getTotalShares();
+    }
+
+    /**
+     * @return the amount of shares that corresponds to `_bnbAmount` protocol-controlled BNB.
+     */
+    function getSharesByPooledBNB(uint256 _bnbAmount) public view returns (uint256) {
+        return (_bnbAmount * _getTotalShares()) / _getTotalPooledBNB();
+    }
+
+    /**
+     * @return the amount of BNB that corresponds to `_sharesAmount` token shares.
+     */
+    function getPooledBNBByShares(uint256 _sharesAmount) public view returns (uint256) {
+        return (_sharesAmount * _getTotalPooledBNB()) / _getTotalShares();
+    }
+
+    /**
+     * @return the total amount (in wei) of BNB controlled by the protocol.
+     * @dev This is used for calculating tokens from shares and vice versa.
+     * @dev This function is required to be implemented in a derived contract.
+     */
+    function _getTotalPooledBNB() internal view virtual returns (uint256);
+
+    /**
+     * @notice Moves `_amount` tokens from `_sender` to `_recipient`.
+     * Emits a `Transfer` event.
+     * Emits a `TransferShares` event.
+     */
+    function _transfer(address _sender, address _recipient, uint256 _amount) internal {
+        require(_sender != address(0), "TRANSFER_FROM_ZERO_ADDR");
+        require(_recipient != address(0), "TRANSFER_TO_ZERO_ADDR");
+        require(_recipient != address(this), "TRANSFER_TO_STBNB_CONTRACT");
+
+        uint256 currentSenderShares = _shares[_sender];
+        require(currentSenderShares >= _amount, "BALANCE_EXCEEDED");
+        unchecked {
```

Comment:
Does `unchecked` required for sol 0.8?

Reply:
I follow the `ERC20Upgradeable.sol` of openzeppelin. It saves gas